react-native.js

```diff
@@ -7,36 +7,48 @@ const ERROR = 2;
 module.exports = {
   extends: [
     require.resolve('./node.js'),
-    'plugin:react/recommended',
-    'prettier/react',
+    require.resolve('./react.js'),
     'plugin:react-native-a11y/all',
   ],
   env: {
     'react-native/react-native': true,
   },
-  plugins: ['react', 'react-native', 'react-hooks'],
-  parserOptions: {
-    ecmaFeatures: {
-      jsx: true,
-    },
-  },
+  plugins: ['react-native'],
   rules: {
-    'react/display-name': OFF,
-    'react/no-multi-comp': [WARNING, { ignoreStateless: true }],
-    'react/no-unused-prop-types': OFF,
-    'react/prop-types': OFF,
-    'react/require-default-props': OFF,
     'react-native/no-unused-styles': ERROR,
     'react-native/split-platform-components': OFF,
     'react-native/no-inline-styles': WARNING,
     'react-native/no-color-literals': WARNING,
     'react-native/no-raw-text': ERROR,
-    'react-hooks/rules-of-hooks': ERROR,
-    'react-hooks/exhaustive-deps': WARNING,
   },
-  settings: {
-    react: {
-      version: 'detect',
+  overrides: [
+    {
+      files: ['*.js', '*.jsx'],
+      settings: {
+        'import/extensions': [...extensions.JS, ...extensions.JS_REACT_NATIVE],
```

Comment:
Same comment as in node.js above. With this configuration, it seems that we do not allow of importing .ts* files from .js* files. Which is exactly the case of default RN typescript template which creates index.js and App.tsx.

Reply:
Oh I thought it's the other way around!

Reply:
Should be good now, passed `.ALL` back